### DISCUSSION TITLE
ConcatDataset & ChainDataset

### DISCUSSION THREAD

**Discussion by vincentAgnus** [Project-MONAI/MONAI#7081] [Ideas]
**Problem**
Torch proposes the ConcatDataset & ChainDataset. I do not find similar in Monai. My typical usage is to use different transforms on a set of images (not necessarily the same) to train my model. The different transforms handle different strategies of data augmentation depending on the image context (in my case object of interest near the border)

**Desired Solution**
The ideal solution is to create these Monai classes to handle the different transforms

** Alternative **
An alternative could be to use  `monai.transforms.OneOf `to randomly choose one transform between all available. In my case is not suitable for my case since transform depends on some image context. My current workaround is to maintain a list of datasets and list of loaders. During the training, I loop over the loaders.

**Additional context**
I work on CT images with 128x128x128 patches

**Comment by wyli** [comment]
thanks for the feature request, these are supported, for example:
```
>>> from monai.data import Dataset
>>> from torch.utils.data.dataset import ConcatDataset
>>> c_ds = ConcatDataset([Dataset([1, 2]), Dataset([2, 3])])
>>> len(c_ds)
4
```
